crates/sui-indexer-alt/src/handlers/mod.rs

```diff
@@ -353,7 +353,7 @@ fn committer<H: Handler + 'static>(
                 }
 
                 // Time to write out another batch of rows, if there are any.
-                _ = poll.tick(), if pending_rows > 0 => {
+                _ = poll.tick(), if rx.is_closed() || pending_rows > 0 || !precommitted.is_empty() => {
```

Comment:
This didn't end up entirely fixing the issue because the receiver can be closed after the `select` starts waiting, which will still cause all of the arms to be unresponsive. In general, it's probably a good rule of thumb to not include conditions on guard arms that can be triggered externally (in this case `rx.is_closed()` is one of those), because they can be have unpredictably.

The fix was to remove all guards from the `poll.tick()` case.